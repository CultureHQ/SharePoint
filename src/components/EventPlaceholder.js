import React from "react";

import styles from "../styles.css";

const EventPlaceholder = () => (
  <div className={styles.event}>
    <div className={styles.eventHeader}>
      <div className={styles.eventLeft}>
        <span className={styles.eventLeftPlaceholder} />
      </div>
      <div className={styles.eventRight}>
        <div className={styles.eventRightPlaceholder} />
        <div className={styles.eventRightPlaceholder} />
        <div className={styles.eventRightPlaceholder} />
      </div>
    </div>
    <div className={styles.eventFooter} />
  </div>
);

export default EventPlaceholder;
