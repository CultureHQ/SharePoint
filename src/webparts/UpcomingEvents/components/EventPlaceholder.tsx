import * as React from "react";

import styles from "../styles.module.scss";

const EventPlaceholder = () => (
  <div className={styles.event}>
    <div className={styles.eventHeader}>
      <div className={styles.eventLeft}>
        <span className={styles.eventLeftPlaceholder} />
      </div>
      <div className={styles.eventRight}>
        <div className={styles.eventRightPlaceholder} />
        <div className={styles.eventRightPlaceholder} />
        <div className={styles.eventRightPlaceholder} />
      </div>
    </div>
    <div className={styles.eventFooter} />
  </div>
);

export default EventPlaceholder;
