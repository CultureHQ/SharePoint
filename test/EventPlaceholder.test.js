import React from "react";
import { shallow } from "enzyme";

import EventPlaceholder from "../src/components/EventPlaceholder";

test("matches snapshot", () => {
  expect(shallow(<EventPlaceholder />)).toMatchSnapshot();
});
