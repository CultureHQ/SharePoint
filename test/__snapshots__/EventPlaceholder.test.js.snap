// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventPlaceholder />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="eventHeader"
        >
          <div
            className="eventLeft"
          >
            <span
              className="eventLeftPlaceholder"
            />
          </div>
          <div
            className="eventRight"
          >
            <div
              className="eventRightPlaceholder"
            />
            <div
              className="eventRightPlaceholder"
            />
            <div
              className="eventRightPlaceholder"
            />
          </div>
        </div>,
        <div
          className="eventFooter"
        />,
      ],
      "className": "event",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="eventLeft"
            >
              <span
                className="eventLeftPlaceholder"
              />
            </div>,
            <div
              className="eventRight"
            >
              <div
                className="eventRightPlaceholder"
              />
              <div
                className="eventRightPlaceholder"
              />
              <div
                className="eventRightPlaceholder"
              />
            </div>,
          ],
          "className": "eventHeader",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="eventLeftPlaceholder"
              />,
              "className": "eventLeft",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "eventLeftPlaceholder",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="eventRightPlaceholder"
                />,
                <div
                  className="eventRightPlaceholder"
                />,
                <div
                  className="eventRightPlaceholder"
                />,
              ],
              "className": "eventRight",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "eventRightPlaceholder",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "eventRightPlaceholder",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "eventRightPlaceholder",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "eventFooter",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="eventHeader"
          >
            <div
              className="eventLeft"
            >
              <span
                className="eventLeftPlaceholder"
              />
            </div>
            <div
              className="eventRight"
            >
              <div
                className="eventRightPlaceholder"
              />
              <div
                className="eventRightPlaceholder"
              />
              <div
                className="eventRightPlaceholder"
              />
            </div>
          </div>,
          <div
            className="eventFooter"
          />,
        ],
        "className": "event",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="eventLeft"
              >
                <span
                  className="eventLeftPlaceholder"
                />
              </div>,
              <div
                className="eventRight"
              >
                <div
                  className="eventRightPlaceholder"
                />
                <div
                  className="eventRightPlaceholder"
                />
                <div
                  className="eventRightPlaceholder"
                />
              </div>,
            ],
            "className": "eventHeader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="eventLeftPlaceholder"
                />,
                "className": "eventLeft",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "eventLeftPlaceholder",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="eventRightPlaceholder"
                  />,
                  <div
                    className="eventRightPlaceholder"
                  />,
                  <div
                    className="eventRightPlaceholder"
                  />,
                ],
                "className": "eventRight",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "eventRightPlaceholder",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "eventRightPlaceholder",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "eventRightPlaceholder",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "eventFooter",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
